### ISSUE TITLE
Can't install on Ubuntu 14.04 LTS VM 

### ISSUE THREAD

**Issue by Tset-Noitamotua** [stopspazzing/Sanic-Server-with-MDL-Blog-Template#2]
Installation with Python 3.4 fails with: 
`RuntimeError: aioodbc doesn't support Python earlier than 3.5`

Installation with Python 3.5 / 3.6 fails with: 
`Failed building wheel for ujson ... fatal error: Python.h: No such file or directory`
`Failed building wheel for httptools ... fatal error: Python.h: No such file or directory`
`Failed building wheel for pyodbc ... fatal error: Python.h: No such file or directory`

**Comment by stopspazzing** [comment]
So the requirements are odd, had same issue as you. You need several packages to be installed for all pip packages to completely install.

So here is a list of packages required:
```
python & python-dev >=3.6
unixodbc
unixodbc-dev
libsqliteodbc
```

On Ubuntu/Linux Mint:
`sudo apt update && sudo apt install unixodbc unixodbc-dev libsqliteodbc`

Now, I installed latest Python 3.6.1 via compiling it with altinstall which is super easy:
https://tecadmin.net/install-python-3-6-ubuntu-linuxmint/

There are several ways of installing Python 3.6, I have found the above way easiest for me.

After which you can use (in project folder): 
`python3.6 -m pip install -r requirements.txt`

NOTE: I personally ran into permission issue with the above and had to use sudo with -H switch; People would probably say this is a bad idea, fyi:
`sudo -H python3.6 -m pip install -r requirements.txt`

**Comment by pyx** [comment]
I would suggest you to use virtual environment, to start, try:
```bash
python3.6 -m venv
```
python3.6 is whatever you installed as the python 3.6 interpreter.

**Comment by stopspazzing** [comment]
Yeah, was just going to update and say env would be smarter, but since all my linux versions are running in VM, yolo. 😈 

I'll update readme with correct information not including the sudo idea, cause you know like bad idea.

**Comment by stopspazzing** [comment]
I have added the required packages to be installed and some other info to the readme. Hopefully that is all that is needed. Going to test it on Ubuntu 14.04 to make sure that satisfies all requirements.

@pyx 
Not sure if that is a given to use a virtual env. Think I should add it? Sanic doesn't mention it.

**Comment by stopspazzing** [comment]
Yeah, just tested it on Ubuntu 14.04. those are the required packages.

**Comment by pyx** [comment]
It is not a hard requirement, for sure.

Sanic is targeting python developers, there is a consensus in using virtualenvs among these people.

I think it is up to you whether to mention it in README, after all, your application's target audiences are different than usual python packages'. The deployment usually involve self-contained environments such as VMs, or docker images, which are in essence virtual environments on steroids.

**Comment by stopspazzing** [comment]
Good point. Thanks @pyx

**Comment by Tset-Noitamotua** [comment]
puh ... I had a hard time to make it (almost) work. Tried to avoid compiling Python3.6 from source first (because I've never did it before) - but all attempts failed e.g. it does not work with the Python 3.6 version that one can get with `apt install python3.6` on Ubuntu 17.04. Finally I did compile it as described in the link you provided - worked at first attempt :)

I say **almost** because allthough I have the server running I now have another [issue](https://github.com/stopspazzing/Sanic-Server-with-MDL-Blog-Template/issues/4)
